public/app/plugins/datasource/loki/logsTimeSplit.ts

```diff
@@ -0,0 +1,49 @@
+// every timestamp in this file is a number which contains an unix-timestamp-in-millisecond format,
+// like returned by `new Date().getTime()`. this is needed because the "math"
+// has to be done on integer numbers.
+
+const MAX_CHUNK_COUNT = 50;
+
+// the way loki handles logs-range-queries is that if you specify start & end,
+// one of those will be included, but the other will not. this allows us to
+// make it easy to split ranges.
+// for example, if the time-range is 100<>150,
+// we can split it into:
+// - 100<>120
+// - 120<>140
+// - 140<>150
+// and no log-line will be skipped or duplicated
+// (NOTE: we do these calculations in milliseconds. at the end, Loki receives
+// nanoseconds, but it will be OK, because it's simply a matter to adding `000000`,
+// to the end, so if we do it right in milliseconds, it should be OK in
+// nanoseconds too
+
+export function getRanges(
```

Comment:
Really minor, but could you rename both `getRanges` functions to something more related with its function. E.g. splitTimeRenge or getRangeChunks.

Reply:
sure, done!